### DISCUSSION TITLE
Add a new dialect for AWS QLDB?

### DISCUSSION THREAD

**Discussion by soulmachine** [sqlalchemy/sqlalchemy#7977] [Usage Questions]
### Describe the use case

AWS QLDB is a managed ledger database that provides a transparent, immutable, and cryptographically verifiable transaction log, I'm wondering whether it is possible to make sqlalchemy support this new database.

It has a driver [pyqldb](https://pypi.org/project/pyqldb/), I'm not sure whether it is compatible with DBAPI.

### Databases / Backends / Drivers targeted

[AWS QLDB](https://aws.amazon.com/qldb/)

### Example Use

SQL:

```sql
SELECT p.* FROM Person AS p BY pid WHERE pid = ?
```

Python:

```python
from pyqldb.driver.qldb_driver import QldbDriver

qldb_driver = QldbDriver(ledger_name="myTest", region_name="us-east-2")

def read_documents(transaction_executor):
    # Transaction started implicitly, no need to explicitly start transaction

    # cursor can be iterated like a standard python iterator
    # to get results
    cursor = transaction_executor.execute_statement("SELECT * FROM Person")

    for doc in cursor:
        print(doc["GovId"])
        print(doc["FirstName"])
```

**Comment by CaselIT** [comment]
Hi,

Sqlalchemy developers don't have the resources to support new databases at the moment.
If you, or anyone, wants to try writing a dialect for that db I've shared some pointers on how to start here https://github.com/sqlalchemy/sqlalchemy/discussions/7968#discussioncomment-2646098

**Comment by zzzeek** [comment]
all new dialects at this point are to be [third party](https://docs.sqlalchemy.org/en/14/dialects/index.html#external-dialects) dialects.  SQLAlchemy no longer adds new dialects to the core library; mysql/sqlite/postgresql/mssql/oracle are it.
